data/templates/secure_erase.py

```diff
@@ -56,6 +65,209 @@
+        # scrub: please verify that device size below is correct!
+        # scrub: scrubbing /dev/sdf1 1995650048 bytes (~1GB)
+        # scrub: random  |................................................|
+        # scrub: random  |................................................|
+        # scrub: 0x00    |................................................|
+        # scrub: verify  |................................................|
+
+        # Maximum dot count for each pass
+        log = open(self.path + drive + '.log', 'r')
+        MAX_DOT_COUNT = 50
+        # Scrub pass counts for different scrub methods, default pass count is 1
+        pass_counts = {"nnsa": 4, "dod": 4, "gutmann": 35, "schneier":7, "pfitzner7":7,
+                       "pfitzner33": 33, "usarmy": 3, "random2": 2, "old": 6, "fastold": 5}
+        pass_count = pass_counts[self.parameters["option"]]
+        line_count = 0
+        dot_count = 0
+        patterna = re.compile("^scrub: \w{4,6}\s*\|\.+\|$")
+        patternb = re.compile("^scrub: \w{4,6}\s*\|\.*$")
+        for line in log.readlines():
+            line = line.strip()
+            if patterna.match(line):
+                line_count += 1
+            elif patternb.match(line):
+                dot_count = len(line.split("|")[1])
+        percent = 100.00*(line_count*MAX_DOT_COUNT + dot_count)/(pass_count*MAX_DOT_COUNT)
+        log.close()
+        return percent
+
+    def __get_hdparm_duration(self, log):
+        """
+        Get hdparm required secure erase time.
+        :param log: a file object of secure erase log
+        :return: required secure erase time indicated by hdparm tool
+        """
+        pattern = re.compile("(\d{1,4})\s*min for SECURITY ERASE UNIT." +
+                             "\s*(\d{1,4})\s*min for ENHANCED SECURITY ERASE UNIT.", re.I)
+        estimated_time = 0
+        for line in log.readlines():
+            line = line.strip()
+            match = pattern.match(line)
+            if match:
+                if self.parameters["option"] == "security-erase":
+                    estimated_time = match.group(1)
+                else:
+                    estimated_time = match.group(2)
+        return float(estimated_time)
+
+    def hdparm_parser(self, drive):
+        """
+        Secure erase job progress parser for scrub tool.
+        :param drive: drive name
+        :return: a float digital of percentage
+        """
+        log = open(self.path + drive + '.log', 'r')
+        percent = 0.0
+        if not self.duration.has_key(drive) or self.duration[drive] == 0:
+            self.duration[drive] = self.__get_hdparm_duration(log)
+        if self.duration[drive] != 0:
+            percent = self.percent + 100.0*self.interval/(self.duration[drive]*60.00)
+            # As hdparm SE duration is estimated, there might be percent larger than 100
+            # Let maximum precent to be 99.00 instead
+            if percent > 100.00:
```

Comment:
Change 100.0 -> 99.0, otherwise there may be progress decrease and confuse user: 60 -> 80 -> 99 -> 99.3 -> 99.5 -> 99